atoma-state/src/state_manager.rs

```diff
@@ -1110,7 +1110,14 @@ impl AtomaState {
             "INSERT INTO stacks 
                 (owner_address, stack_small_id, stack_id, task_small_id, selected_node_id, num_compute_units, price, already_computed_units, in_settle_period, total_hash, num_total_messages) 
                 VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)
-                ON CONFLICT (stack_small_id) DO NOTHING",
+                ON CONFLICT (stack_small_id) DO UPDATE
+                SET already_compute_units = 
+                    CASE 
+                        WHEN stacks.already_computed_units + $8 <= stacks.num_compute_units
+                        THEN stacks.already_computed_units + $8
+                        ELSE RAISE EXCEPTION 'Computed units would exceed total compute units'
+                    END
+                WHERE stacks.already_computed_units + $8 <= stacks.num_compute_units",
```

Comment:
this is good, one more thing and then we are good
during the first insert you can request have `already_computed_units` greater than `num_compute_units` which is bad
we can solve this by adding `constraint` to the table, and then we could also simplify the update query, because the constraint should take care of that 
the constraint should be `CONSTRAINT check_compute_units CHECK (already_computed_units <= num_compute_units)`

Reply:
Thanks for the suggestion, I addressed it the way you propose